nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2672,6 +2676,25 @@ public DocumentModel getOrCreateDocument(DocumentModel docModel,
         });
     }
 
+    @Override
+    public DocumentModel newDocumentModel(DocumentRef parentRef, String name, String typeName) {
+        if (parentRef == null && name == null) {
+            return createDocumentModel(typeName);
+        }
+
+        String parentPath = null;
+        if (parentRef != null) {
+            parentPath = resolveReference(parentRef).getPath();
+        }
+        Map<String, Serializable> options = new HashMap<>();
+        options.put(CoreEventConstants.PARENT_PATH, parentPath);
```

Comment:
We should audit our code for usage of `PARENT_PATH` because as this code shows it's now meaningless in some cases so we may as well try avoid using it completely (and replace it with something else depending on needs)

Reply:
Hm actually it's even worse, for the `moveDocument` case the value is a `DocumentRef` and not a `String`... and this is used by quotas...

Reply:
yes I agree :(